arangod/Auth/User.cpp

```diff
@@ -584,7 +584,7 @@ auth::Level auth::User::collectionAuthLevel(std::string const& dbname,
     if (it != _dbAccess.end()) {
       // Second try to find a specific grant
       CollLevelMap::const_iterator pair =
-          it->second._collectionAccess.find(cname);
+          it->second._collectionAccess.find(std::string{cname});
```

Comment:
can we save this conversion to string somehow? I guess that the authentication code is called quite often, so it would make sense to make it as quick as possible.

Reply:
Right now I do not know why, unless we change the data structure of: 
`typedef std::unordered_map<std::string, auth::Level> CollLevelMap;` as well.

Reply:
What if we give this map a customer hash and eq function? Then we should be able to use heterogeneous lookups.

Reply:
Added in abd4ec43b7044cda33327501d291e8cf3f106864